integration-tests/testkit/test.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * This module uses Vitest's fixture system to make common usage patterns
+ * of our testkit easily consumable in test cases. @see https://vitest.dev/guide/test-context.html#test-extend
+ */
+
+import { test as testBase } from 'vitest';
+import { createTmpFileController, TmpFileController } from './fs';
+
+interface Context {
+  graphqlFile: TmpFileController;
+}
+
+export const test = testBase.extend<Context>({
+  graphqlFile: async ({}, use) => {
```

Comment:
How is this different from:

```ts
const tmpFile = createTmpFile({ extension: 'graphql' });
```

and how does it improve the testing?

Reply:
The way to discover `graphqlFile` in the `test` function arguments is even harder then looking at `testkit/**` and trying to find something that already exists and may be useful.

Reply:
Also, `graphqlFile` is always created and can't be reused

Reply:
> Also, graphqlFile is always created and can't be reused (can't create another temporary graphql file)

I don't understand this comment. Fixtures are created for each test they are used in. So that would suggest you're wrong. Am I missing something?

Reply:
Try to use 2 temporary files in a test

Reply:
We would have to redo the fixture. That’s why passing a function to create a temporary file is imo future proof

Reply:
> Try to use 2 temporary files in a test

That's a different use-case. Since you brought it up, I will mention that one pattern I like for tests that need to run against a project is a project instance fixture that, when used, creates a temporary directory from which all other methods are relative to. Here is an example https://github.com/graffle-js/graffle/blob/main/tests/e2e/e2e.test.ts#L20-L44. It also shows using a service bound to a test.

But this is getting off topic.